compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -1254,6 +1254,14 @@ OMR::ResolvedMethodSymbol::genIL(TR_FrontEnd * fe, TR::Compilation * comp, TR::S
                comp->getOption(TR_EnableOSR) && comp->supportsInduceOSR() &&
                !comp->isPeekingMethod()  && (comp->getOption(TR_EnableNextGenHCR) || !comp->getOption(TR_EnableHCR)) ;
 
+            if (!comp->isPeekingMethod())
+               {
+               if (comp->getOSRMode() == TR::involuntaryOSR && !doOSR)
```

Comment:
This catches a general case, but isn't there a more specific check - namely we must support OSR for every node which returns true to isPotentialOSRPoint and shouldn't that check be enforced during ilgen? I suspect genTreeTop or similar may be the place we want to do something. This is only going to pick-up on global incompatibility not individual sites...